# Portfolio Task 2 
    companion note book

### NAME : ASHRAF ALI KHAN PATAN

### Data Set Info :
- The data set is downloaded from Kaggle .
- This data set shows various key attributes and their significance.
- Link for Data Set https://www.kaggle.com/datasets/swarajkhan/wood-prices-dataset
- This data set was uploaded in the year 2023.

- Importing Packages

In [1]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import geopandas as gpd
import plotly.io as pio
from shapely.geometry import Point

- Loading Data set 

In [2]:
wood_df = pd.read_csv('wood_prices_dataset.csv')

- Displaying first 5 rows of the data set

In [3]:
wood_df.head()

Wood Type       Country  Price (USD) Supply Source Quality Rating  \
0     Maple         China       166.59      Imported         Medium   
1  Rosewood  South Africa       144.71         Local            Low   
2  Rosewood  South Africa       216.92         Local         Medium   
3       Oak        Russia       130.13      Imported         Medium   
4    Bamboo     Australia       114.66         Local         Medium   

   Popularity Demand Level Availability  
0           3       Medium     Moderate  
1           6       Medium     Moderate  
2           2       Medium     Abundant  
3           5          Low     Abundant  
4           8          Low     Abundant

- Displaying last 5 rows of the data set

In [14]:
wood_df.tail()

Wood Type       Country  Price (USD) Supply Source Quality Rating  \
995  Rosewood     Indonesia       151.81         Local         Medium   
996      Pine       Germany       230.97      Imported            Low   
997     Maple     Indonesia       200.87         Local            Low   
998     Maple  South Africa       227.75      Imported            Low   
999  Mahogany     Indonesia       122.71      Imported            Low   

     Popularity Demand Level Availability  
995           2       Medium      Limited  
996           4          Low     Abundant  
997           4         High      Limited  
998           8         High     Moderate  
999           5          Low      Limited

In [12]:
wood_df.columns

Index(['Wood Type', 'Country', 'Price (USD)', 'Supply Source',
       'Quality Rating', 'Popularity', 'Demand Level', 'Availability'],
      dtype='object')

Describing features in the Data Set :

Wood Type: types of wood possess characteristics and applications. For example Maple is recognized for its strength and longevity making it a preferred choice, for flooring and furniture whereas Bamboo is valued for its nature and rapid growth leading to its popularity in friendly goods.

Country: The origin country significantly influences the cost and quality of the wood. Factors such as forestry laws, labor expenses and the presence of established supply chains can impact these attributes. For instance countries like Brazil and Indonesia are known for their forest reserves while nations like Japan may prioritize importing high quality wood due to local resources.

Price in USD; Having a grasp of the woods price in USD facilitates comparisons across markets. Enables informed purchasing decisions. Prices can vary based on factors such as wood type, quality, sourcing location and market demand.

Supply Source: Whether the wood is sourced locally or imported can affect its cost and availability. Imported wood may incur expenses related to shipping, tariffs and import taxes whereas sourced wood could be subject to supply constraints or regulatory measures.

Quality Rating: The grade of wood impacts its suitability, for uses.
Good quality timber is usually pricier. Favored for luxury items while lower quality timber could be used in mass produced goods.

Popularity: The popularity of a type of wood can affect its demand and cost. Highly sought after wood varieties may command prices and demand reflecting consumer preferences and market trends.

Demand Level: Knowing the level of demand helps in understanding market dynamics. Timber, with demand may face price hikes and supply shortages whereas less, in demand timber may be easily accessible and affordable.

Availability: The availability of timber influences market stability and pricing. Abundant types of wood generally have prices while scarce varieties may see significant price fluctuations.


### Data Pre-Processing

1. Type and Shape of the Data Set 

In [6]:
print(f'Dataset type: {type(wood_df)}')         
print(f'Dataset shape: {wood_df.shape}')   

(1000, 8)

- Data Set contains 1000 rows and 8 columns

2. Information about the data set 

In [4]:
wood_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Wood Type       1000 non-null   object 
 1   Country         1000 non-null   object 
 2   Price (USD)     1000 non-null   float64
 3   Supply Source   1000 non-null   object 
 4   Quality Rating  1000 non-null   object 
 5   Popularity      1000 non-null   int64  
 6   Demand Level    1000 non-null   object 
 7   Availability    1000 non-null   object 
dtypes: float64(1), int64(1), object(6)
memory usage: 62.6+ KB


- This dataset contains 1,000 entries and 8 columns related to wood types. Each column represents a different attribute:
    - Wood Type: The type or species of wood.
    - Country: The country where the wood is from.
    - Price (USD): The price of the wood in US dollars.
    - Supply Source: Where the wood is sourced from.
    - Quality Rating: The quality rating of the wood (e.g., high, medium, low).
    - Popularity: A numerical value indicating how popular the wood is.
    - Demand Level: The level of demand for the wood (e.g., high, medium, low).
    - Availability: How readily available the wood is (e.g., abundant, scarce).
- There are a variety of data types: most columns are categorical, while the price is a float type, and popularity is a integer type . 
- The total memory usage of this dataset is about 62.6 kilobytes.

3. Describe Method :

In [7]:
wood_df.describe()

Price (USD)   Popularity
count  1000.000000  1000.000000
mean    165.745780     5.526000
std      48.427786     2.859983
min      80.230000     1.000000
25%     123.485000     3.000000
50%     165.265000     5.500000
75%     207.977500     8.000000
max     249.930000    10.000000

- The describe() method is used to perfom statistical calculations such as mean, standard deviation and min-max values

4. Checking Null Values

In [13]:
null_counts = wood_df.isnull().sum()
print(null_counts)

Wood Type         0
Country           0
Price (USD)       0
Supply Source     0
Quality Rating    0
Popularity        0
Demand Level      0
Availability      0
dtype: int64


- The dataset is complete with no missing values

## Use of AI 

- I have used perpelxity.ai for handling the errors.
- Reference link https://www.perplexity.ai 

### Usage of Other Applications

- I have used youtube for learning new graphs such as Tree maps and Sunburst map.